public/app/ui/components/search/queryBuilder/QueryBuilderComponent.less

```diff
@@ -0,0 +1,74 @@
+.query-builder {
+    .group-conditions {
+        margin: @padding-small-vertical 0 @padding-small-vertical 30px;
+    }
+
+    .rules-list {
+        padding: 0 0 0 30px;
+        position: relative;
+
+        > * {
```

Comment:
These are the lines in the query builder that connect the and/or switch to the conditions and groups that are subordinate to it.  I won't explain it all here, but essentially the `::before` part is the line down and across, and the `::after` part is the line below the line that goes across, to the top of the next one.  (And `::after` is hidden on the `:last-child`).